### PR TITLE
Backport rain config

### PR DESCRIPTION
**Pull request by Asherslab**
# Changes proposed in this pull request:
- Backports the rain config option
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/Structures.java [hunk 1]
```diff
@@ -60,200 +117,863 @@ public static void init()
...
+        }
+        return new ArrayList<>();
     }
 
-    private static void addDecorationStyle(final String decoration, final String style)
+    /**
+     * Class to handle schematic naming.
+     * It does extract information from a schematic using its name.
+     */
+    public static class StructureName
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Structures.java]: MAJOR Reduce this class from 118 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/colony/Structures.java [hunk 2]
```diff
@@ -60,200 +117,863 @@ public static void init()
...
+            try (FileSystem fileSystem = FileSystems.getFileSystem(uri))
             {
-                decorationFolder = new File(FMLCommonHandler.instance().getMinecraftServerInstance().getDataDirectory(), "minecolonies/decorations");
+                final Path basePath = fileSystem.getPath(SCHEMATICS_ASSET_PATH);
+                Log.getLogger().info("Load huts or decorations from jar");
+                loadSchematicsForPrefix(basePath, SCHEMATICS_PREFIX);
             }
-
-            if (!decorationFolder.exists() && !decorationFolder.mkdirs())
+            catch (@NotNull IOException | FileSystemNotFoundException e1)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Structures.java]: MAJOR Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)

File: src/main/java/com/minecolonies/coremod/colony/Structures.java [hunk 3]
```diff
@@ -60,200 +117,863 @@ public static void init()
 
     /**
      * Loads all styles saved in ["/assets/minecolonies/schematics/"].
-     * Puts these in {@link #hutStyleMap}, with key being the name of the hutDec (E.G. Lumberjack).
-     * and the value is a list of styles. Puts decorations in {@link #decorationStyleMap}.
+     * Puts these in {@link #md5Map}, with key being the fullname of the structure (schematics/stone/Builder1).
      */
+    // The same exception will be triggered in the 2nd catch with logging this time.
+    @SuppressWarnings("squid:S1166")
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Structures.java]: INFO Suppressing the 'squid:S1166' warning is not allowed [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 4]
```diff
@@ -83,24 +116,25 @@
...
-        addMapping("Miner", BuildingMiner.class, BlockHutMiner.class);
-        addMapping("Stonemason", BuildingStonemason.class, BlockHutStonemason.class);
-        addMapping("TownHall", BuildingTownHall.class, BlockHutTownHall.class);
-        addMapping("Warehouse", BuildingDeliveryman.class, BlockHutDeliveryman.class);
-        addMapping("Fisherman", BuildingFisherman.class, BlockHutFisherman.class);
-        addMapping("GuardTower", BuildingGuardTower.class, BlockHutGuardTower.class);
+        addMapping("Baker", BuildingBaker.class, BuildingBaker.View.class, BlockHutBaker.class);
+        addMapping("Blacksmith", BuildingBlacksmith.class, BuildingBlacksmith.View.class, BlockHutBlacksmith.class);
+        addMapping("Builder", BuildingBuilder.class, BuildingBuilderView.class, BlockHutBuilder.class);
+        addMapping("Home", BuildingHome.class, BuildingHome.View.class, BlockHutCitizen.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingHome". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingHome". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 5]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Farmer", BuildingFarmer.class, BuildingFarmer.View.class, BlockHutFarmer.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingFarmer". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingFarmer". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 6]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Lumberjack", BuildingLumberjack.class, BuildingLumberjack.View.class, BlockHutLumberjack.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingLumberjack". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingLumberjack". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 7]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Miner", BuildingMiner.class, BuildingMiner.View.class, BlockHutMiner.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingMiner". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingMiner". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 8]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Stonemason", BuildingStonemason.class, BuildingStonemason.View.class, BlockHutStonemason.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingStonemason". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingStonemason". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 9]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("TownHall", BuildingTownHall.class, BuildingTownHall.View.class, BlockHutTownHall.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingTownHall". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingTownHall". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 10]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Deliveryman", BuildingDeliveryman.class, BuildingDeliveryman.View.class, BlockHutDeliveryman.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingDeliveryman". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingDeliveryman". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 11]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Fisherman", BuildingFisherman.class, BuildingFisherman.View.class, BlockHutFisherman.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingFisherman". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingFisherman". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 12]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("GuardTower", BuildingGuardTower.class, BuildingGuardTower.View.class, BlockHutGuardTower.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingGuardTower". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingGuardTower". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 13]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("WareHouse", BuildingWareHouse.class, BuildingWareHouse.View.class, BlockHutWareHouse.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingWareHouse". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingWareHouse". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 14]
```diff
@@ -1,41 +1,58 @@
...
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
+
+import static com.minecolonies.api.util.constants.ToolLevelConstants.TOOL_LEVEL_HAND;
 
 /**
  * Base building class, has all the foundation for what a building stores and does.
  */
 public abstract class AbstractBuilding
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 34 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
MAJOR class "AbstractBuilding" has 60 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 15]
```diff
@@ -83,24 +116,25 @@
...
-        addMapping("Home", BuildingHome.class, BlockHutCitizen.class);
-        addMapping("Farmer", BuildingFarmer.class, BlockHutFarmer.class);
-        addMapping("Lumberjack", BuildingLumberjack.class, BlockHutLumberjack.class);
-        addMapping("Miner", BuildingMiner.class, BlockHutMiner.class);
-        addMapping("Stonemason", BuildingStonemason.class, BlockHutStonemason.class);
-        addMapping("TownHall", BuildingTownHall.class, BlockHutTownHall.class);
-        addMapping("Warehouse", BuildingDeliveryman.class, BlockHutDeliveryman.class);
-        addMapping("Fisherman", BuildingFisherman.class, BlockHutFisherman.class);
-        addMapping("GuardTower", BuildingGuardTower.class, BlockHutGuardTower.class);
+        addMapping("Baker", BuildingBaker.class, BuildingBaker.View.class, BlockHutBaker.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingBaker". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingBaker". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 16]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Blacksmith", BuildingBlacksmith.class, BuildingBlacksmith.View.class, BlockHutBlacksmith.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingBlacksmith". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
MAJOR Remove this reference to "BuildingBlacksmith". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 17]
```diff
@@ -83,24 +116,25 @@
...
+        addMapping("Builder", BuildingBuilder.class, BuildingBuilderView.class, BlockHutBuilder.class);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MAJOR Remove this reference to "BuildingBuilder". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+    
+    /**
+     * Convert the given metadata into a BlockState for this Block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState getStateFromMeta(final int meta)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 19]
```diff
@@ -0,0 +1,278 @@
...
+        return this.getDefaultState().withProperty(FACING, enumfacing);
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SideOnly(Side.CLIENT)
+    @Override
+    @Deprecated
+    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 20]
```diff
@@ -0,0 +1,278 @@
...
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 21]
```diff
@@ -0,0 +1,278 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 22]
```diff
@@ -0,0 +1,278 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 23]
```diff
@@ -0,0 +1,278 @@
...
+    }
+
+    /**
+     * initialize the block
+     * sets the creative tab, as well as the resistance and the hardness.
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: MINOR Define the locale to be used in this String operation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 24]
```diff
@@ -0,0 +1,278 @@
...
+        setLightOpacity(LIGHT_OPACITY);
+        blockMaterial.setReplaceable();
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getCollisionBoundingBox(IBlockState blockState, World worldIn, BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 25]
```diff
@@ -565,43 +846,59 @@ public void onServerTick(@NotNull final TickEvent.ServerTickEvent event)
     /**
      * Update Subscribers with Colony, Citizen, and AbstractBuilding Views.
      */
-    public void updateSubscribers()
+    private void updateSubscribers()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MAJOR Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 26]
```diff
@@ -957,7 +1424,7 @@ public void spawnCitizen(final CitizenData data)
                 if (getMaxCitizens() == getCitizens().size())
                 {
                     //TODO: add Colony Name prefix?
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 27]
```diff
@@ -441,44 +654,112 @@ public int getID()
...
-     * Increment the mobs killed by this colony.
-     * <p>
-     * Will award achievements for mobs killed.
+     * Get the amount of statistic.
+     *
+     * @param statistic the statistic.
+     * @return amount of statistic.
      */
-    public void incrementMobsKilled()
+    private int getStatisticAmount(@NotNull String statistic)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MAJOR Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 28]
```diff
@@ -643,6 +947,7 @@ public void updateSubscribers()
         isBuildingsDirty = false;
         permissions.clearDirty();
 
+        // TODO: look into potentially keeping lists as well
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java [hunk 29]
```diff
@@ -464,16 +841,85 @@ public int getElementCount()
...
+                final Action action = Action.values()[actionIndex];
+                final String name = LanguageHandler.format(KEY_TO_PERMISSIONS + action.toString().toLowerCase());
+
+                if(name.contains(KEY_TO_PERMISSIONS))
+                {
+                    return;
+                }
+
+                rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(name);
+                final boolean isTriggered = townHall.getColony().getPermissions().hasPermission(Rank.valueOf(actionsList.getParent().getID().toUpperCase()), action);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java]: MINOR Define the locale to be used in this String operation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java [hunk 30]
```diff
@@ -464,16 +841,85 @@ public int getElementCount()
...
         });
     }
 
     /**
+     * Fills the permission list in the GUI.
+     */
+    private void fillPermissionList(@NotNull final String category)
+    {
+        actionsList = findPaneOfTypeByID(LIST_ACTIONS + category, ScrollingList.class);
+        actionsList.setDataProvider(new ScrollingList.DataProvider()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java]: MAJOR Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java [hunk 31]
```diff
@@ -0,0 +1,230 @@
...
+        //todo add specialized view for the warehouse later.
+        @NotNull
+        @Override
+        public Window getWindow()
+        {
+            return new WindowWareHouseBuilding(this);
+        }
+
+        @Override
+        public void deserialize(@NotNull final ByteBuf buf)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java]: MINOR Remove this method to simply inherit it. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1185)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java [hunk 32]
```diff
@@ -0,0 +1,230 @@
...
+                    tileEntity.setBuilding(this);
+                }
+            }
+        }
+
+        return tileEntity;
+    }
+
+    @Override
+    public void serializeToView(@NotNull final ByteBuf buf)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java]: MINOR Remove this method to simply inherit it. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1185)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java [hunk 33]
```diff
@@ -0,0 +1,230 @@
...
+    @Override
+    public String getSchematicName()
+    {
+        return WAREHOUSE;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java [hunk 34]
```diff
@@ -0,0 +1,230 @@
...
+         *
+         * @param c the colonyview to put it in
+         * @param l the positon
+         */
+        public View(final ColonyView c, final BlockPos l)
+        {
+            super(c, l);
+        }
+
+        //todo add specialized view for the warehouse later.
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingDeliveryman.java [hunk 35]
```diff
@@ -43,7 +64,7 @@ public String getSchematicName()
     @Override
     public int getMaxBuildingLevel()
     {
-        return 4;
+        return 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingDeliveryman.java]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingDeliveryman.java [hunk 36]
```diff
@@ -64,17 +85,12 @@ public AbstractJob createJob(final CitizenData citizen)
     public void readFromNBT(@NotNull final NBTTagCompound compound)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingDeliveryman.java]: MINOR Remove this method to simply inherit it. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1185)

**Comment by xavierh** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1159/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 48 issues
* ![MAJOR][MAJOR] 15 major
* ![MINOR][MINOR] 30 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.296 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L65](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L65): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 56 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L65](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L65): class "EntityCitizen" has 94 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L350](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L350): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L351](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L351): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L351](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L351): Assign this magic number 0.02F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L352](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L352): Assign this magic number 0.6D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Assign this magic number 6.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L382](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L382): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L593](https://github.com/Minecolonies/minecolonies/blob/e2e63dffb5ba6b92059ef84828bc1631e1079dc1/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L593): Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1159/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**